tf/cloud-functions-gen2/cloud-function.tf

```diff
@@ -0,0 +1,100 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+resource "google_cloudfunctions2_function" "function" {
+  name     = "e2etest-${terraform.workspace}"
+  location = "us-central1"
+
+  timeouts {
+    create = "5m"
+  }
+
+  build_config {
+    runtime     = var.runtime
+    entry_point = var.entrypoint
+    source {
+      storage_source {
+        bucket = google_storage_bucket.bucket.name
+        object = google_storage_bucket_object.object.name
+      }
+    }
+  }
+
+  service_config {
+    max_instance_count = 1
+    available_memory   = "256M"
+    timeout_seconds    = 120
+    environment_variables = {
+      # Environment variables available during function execution
+      "PROJECT_ID"          = var.project_id
+      "RESPONSE_TOPIC_NAME" = module.pubsub.info.response_topic.topic_name
+    }
+    ingress_settings               = "ALLOW_INTERNAL_ONLY"
+    all_traffic_on_latest_revision = true
+  }
+
+  event_trigger {
+    trigger_region = "us-central1"
+    event_type     = "google.cloud.pubsub.topic.v1.messagePublished"
+    pubsub_topic   = google_pubsub_topic.topic.id
```

Comment:
Why not just use `module.pubsub.info.request_topic.topic_name`?

Reply:
So for some reason, I'm not entirely sure what, this did not work (this was what I tried first). 

But there might be some other reason for it (there was a lot of hit & try). I will retry this approach for re-using the topic once.

Reply:
So tried it again, did not work - function creation fails with this error: 
```
Error 400: Cannot create trigger projects/opentelemetry-ops-e2e/locations/us-central1/triggers/e2etest-84fd6dc1938c-340863: The request was invalid: pubsub topic request-84fd6dc1938c is invalid or in a different project
```

This is consistent with what I saw before.